### PR TITLE
Gateway 2827 cherry pick

### PR DESCRIPTION
**Pull request by alameluchidambaram**
Merge 3.3.1.3 fixes into 4.0. This Pull Request is only for PD Service

### CODE REVIEW

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/nhin/proxy/NhinPatientDiscoveryProxyWebServiceSecuredImpl.java [hunk 1]
```diff
@@ -97,11 +97,12 @@ public PRPAIN201306UV02 respondingGatewayPRPAIN201305UV02(PRPAIN201305UV02 reque
                 if (NullChecker.isNotNullish(url)) {
                     ServicePortDescriptor<RespondingGatewayPortType> portDescriptor = new RespondingGatewayServicePortDescriptor();
                     CONNECTClient<RespondingGatewayPortType> client = CONNECTClientFactory.getInstance()
-                            .getCONNECTClientSecured(portDescriptor, url, assertion);
+                            .getCONNECTClientSecured(portDescriptor, assertion, url, target.getHomeCommunity().getHomeCommunityId(), 
+                                    NhincConstants.PATIENT_DISCOVERY_SERVICE_NAME);
 
-                    oProxyHelper.addTargetCommunity((BindingProvider) client.getPort(), target);
+                  /*  oProxyHelper.addTargetCommunity((BindingProvider) client.getPort(), target);
```
Comment (akong) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/nhin/proxy/NhinPatientDiscoveryProxyWebServiceSecuredImpl.java:103]: Remove commented out code.

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/OutboundPatientDiscoveryFactory.java [hunk 2]
```diff
@@ -49,11 +49,9 @@ private OutboundPatientDiscoveryFactory() {
     public OrchestrationContextBuilder createOrchestrationContextBuilder(NhincConstants.GATEWAY_API_LEVEL apiLevel) {
         log.debug("EntityPatientDiscoveryFactory has apiLevel=" + apiLevel.toString());
         switch (apiLevel) {
-        case LEVEL_g0:
```
Comment (akong) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/entity/OutboundPatientDiscoveryFactory.java:52]: There's really no need for the switch statement here as you're always returning the same builder.  Also, line 53 shouldn't be a log warning.

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/nhin/proxy/NhinPatientDiscoveryProxyWebServiceSecuredImpl.java (hunk 1 above)
Comment (alameluchidambaram) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/nhin/proxy/NhinPatientDiscoveryProxyWebServiceSecuredImpl.java:103]: will remove it

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/mpi/adapter/proxy/AdapterMpiProxyWebServiceSecuredImpl.java [hunk 3]
```diff
@@ -80,11 +82,17 @@ protected Log createLogger() {
...
-            ServicePortDescriptor<AdapterMpiSecuredPortType> portDescriptor, String url, AssertionType assertion) {
+            ServicePortDescriptor<AdapterMpiSecuredPortType> portDescriptor, String url, AssertionType assertion) throws PropertyAccessException {
+        
+        String targetHomeCommunityId = null;
+        targetHomeCommunityId = PropertyAccessor.getInstance().getProperty(NhincConstants.GATEWAY_PROPERTY_FILE, 
+                NhincConstants.HOME_COMMUNITY_ID_PROPERTY);
 
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor, url, assertion);
+        return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor, assertion, url,
+                targetHomeCommunityId, NhincConstants.PATIENT_DISCOVERY_ACTION);
```
Comment (akong) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/mpi/adapter/proxy/AdapterMpiProxyWebServiceSecuredImpl.java]: Are we supposed to pass in that constant or the PD service name?  I see the other calls passing in the service name.

**Comment by akong** [comment]
After discussion with Alamelu, we decided that adding the target community id and the service name (or PurposeOfForUseDecider to use only make sense on the Nhin layer (similar to the fix for 3.3.1.3) so we should not be passing those to secured endpoint on the adapter layer.

**Pull request merged**